server/s3_object_data_copier.cc

```diff
@@ -189,19 +187,20 @@ void S3ObjectDataCopier::write_data_block() {
   s3_log(S3_LOG_INFO, request_id, "%s Entry\n", __func__);
 
   assert(!write_in_progress);
-  assert(data_blocks_writing.empty());
+
   assert(!data_blocks_read.empty());
 
-  data_blocks_writing = std::move(data_blocks_read);
+  assert(p_evbuffer_read != nullptr);
+  assert(!p_evbuffer_write);
+  p_evbuffer_write = p_evbuffer_read;
```

Comment:
If I get this correct, you are passing evbuffers of read to write

Reply:
Not exactly. These variables are used only to hold evbuffer's and to free them in timely manner.